### ISSUE TITLE
Introduce duplicate merge option for infrastructure data via bulk edit mode

### ISSUE THREAD

**Issue by kwa20** [SORMAS-Foundation/SORMAS-Project#5888]
### Situation Description

Infrastructure data can neither be deleted nor can it be merged if there are any duplicates. The only way to make it disappear from drop-downs in entities is to archive excess or old entries. However, introducing merge views for respective infrastructure data seems unnecessarily effortful and worthwhile.
### Feature Description

Introduce a bulk action option to merge any type of infrastructure duplicates by adding a new bulk action called "... Merge/Pick"
- This action should only work if two entries are selected and disabled if the selected number equals >2.
- If selected, open a pop-up which displays the selected data side by side (non-editable)
    - Add check boxes to allow selecting either one or the other as the leading entry. Only one checkbox should be able to selected at a time. Selecting the one automatically deselects the other.
    - Provide options to cancel, merge or pick (merge and pick should only get enabled when one entry is selected as leading
    - merge: the leading entry remains and is updated with data from the entry to be discarded. All existing references for the discarded entry should be updated with the remaining one.
    - pick: the leading entry remains without updates and the other entry is discarded. All existing references for the discarded entry should be updated with the remaining one.
### Additional Information

Example (action should be "... Merge/Pick":
<img width="679" alt="infrastructureMerge2" src="https://user-images.githubusercontent.com/70317594/122786735-e736b300-d2b4-11eb-8228-eaca337b26fc.PNG">

Example (entries should completely be non-editable):
<img width="658" alt="infrastructureMerge" src="https://user-images.githubusercontent.com/70317594/122786742-e9007680-d2b4-11eb-8cb2-6321702e8490.PNG">

**Comment by bernardsilenou** [comment]
I think most urgent need is to prevent the importation/ creation of duplicate infrastructure data in the first place #6089.
For systems that have duplicates, we can archive and manually migrate entities to the other infrastructure data.

**Comment by MartinWahnschaffe** [comment]
* This would be relevant for Continent, Subcontinent, Country, Area, Region, District, Community, Facility and PointOfEntry
* Should be implemented in a generic way based on InfrastructureType. This will probably mean that we need to have a generic getEditComponent method in InfrastructureController

@kwa20 
With the few fields that all those entities have, pick/merge does not bring a lot of benefit and may confuse users. Wouldn't it be better to allow the user to edit the picked entity, so data can be adjusted or copied & pasted where needed?

**Comment by JonasCir** [comment]
cc me

**Comment by kwa20** [comment]
@MartinWahnschaffe Good point. Yes, I think that makes sense. I guess we then have to update the infrastructure either with the unedited chosen entry or both previous entries with the picked and edited version. An info text describing consequences of this action might be useful then.

@JonasCir Would this feature even make sense for those systems that are configured with S2S? I guess the difference would be that as a user, you should only be able to merge old infrastructure data with the valid central infrastructure data.

**Comment by bernardsilenou** [comment]
@MateStrysewske @kwa20 
I just got some thoughts that raise some point to consider, maybe its obvious since the current method does not allow archiving regions when all districts in it have not been archived.
When me merge two regions e.g. A = BS- Braunchweig and B = Braunschweig
1. Then the we need to merge all associated districts and facilities and POE in both regions:
   1.1 one  facility for example that is resisted as duplicate in region A and in B could have the same name or different names
      1.1.1  If it is the same name, then we can merge automatically, this is fine.
      1.1.2  if the names are not the same, then the user has to decide if we keep them as separate or merge them, right?
If point 1.1.2 is not completed for all districts, communities and facilities, then we do not merge the regions. In real life, we can have hundred of facilities in a region, so would users be able to put in the time to first merge all facilities and districts in order to be able to merge the regions ? 
2. If jurisdictions have been merged and while merging, the data of the jurisdiction retained (eg region A) is edited. I assume after the merge, the jurisdiction that is not retained is either archived or deleted, thus non considered in all logics in the system.
   2.1 If entities existed on the mobile app based on region B, how are we going to handle this during the synch process?
     2.1.1 If archived, then is it good to continue synching entities to an archived jurisdiction? Or We need to additionally check if the jurisdiction of the mobile app in the same by server each time we synhc and then transfer all data from mobile device to their respective new jurisdiction each time we detect that that the jurisdiction mobile app is not the same as web + update the jurisdiction on the mobile app. 
     2.1.2 If deleted, then do we need further methods to contact the history db to determine where to push them, which may affect  synch performance?
